components/raftstore/src/store/peer.rs

```diff
@@ -4763,23 +4769,57 @@ where
         false
     }
 
+    pub fn set_pending_transfer_leader_msg(&mut self, msg: &eraftpb::Message) {
+        // log_term is set by original leader in pre transfer leader stage.
+        // Callers must guarantee that the message is a valid transfer leader
+        // message.
+        //
+        // See more in `Peer::pre_transfer_leader`.
+        assert!(
+            msg.get_msg_type() == eraftpb::MessageType::MsgTransferLeader
+                && msg.get_log_term() == self.term(),
+            "{} unexpected message type {:?}",
+            self.tag,
+            msg.get_msg_type(),
+        );
+        self.transfer_leader_state.transfer_leader_msg = Some(msg.clone());
+    }
+
+    pub fn maybe_ack_transfer_leader_msg<T>(&mut self, ctx: &mut PollContext<EK, ER, T>) -> bool {
```

Comment:
Pls add some necessary comments here to specify that after the follower has warmed up the cache (>= low_index), it will return ack to the leader and continues on leader transferring.